### DISCUSSION TITLE
Generating cracks in moose for phase field mehtod using gmsh

### DISCUSSION THREAD

**Discussion by ayush4424** [idaholab/moose#27760] [Q&A Meshing]
I have attached an image(ideal crack). How can i generate a crack like this in moose or using gmsh for my phase field applications. When i try by using side set generator and using dirichlet boundary condition of the phase field variable in that region equal to one, the crack is never smooth and i don't get correct load vs reaction graphs for Miehe Tension test. The second image is how i generate a crack.
Please guide how to model a crack in moose correctly
![ideal crack](https://github.com/idaholab/moose/assets/135033854/79382dc1-9b12-4eae-8d4a-40b8e0a35f8e)
![Screenshot from 2024-05-31 17-36-50](https://github.com/idaholab/moose/assets/135033854/8baeb8af-8694-448a-9fb3-7a7e04af616f)

**Comment by GiudGiud** [comment]
Hello

You can assign different subdomain to the top and bottom part of the mesh
then use the BreakMeshByBlockGenerator to split the mesh.

The "Break"Mesh generators are generally meant for this purpose

Guillaume
